### DISCUSSION TITLE
Battle log auto-hide

### DISCUSSION THREAD

**Discussion by Districh-ru** [ihhub/fheroes2#6420] [Ideas]
On the battlefield, if the player opens the battle log, it remains on the screen until the player closes it. But if the player performs an action or skips a turn, he cannot close battle log until his next turn, so the battle log can hide some enemy action or the effect of some mass spells behind it:

https://user-images.githubusercontent.com/113276641/209578582-a3c650e2-6f16-4f82-84bb-23fd245f5e3b.mp4

We could implement the battle log auto-hide before any action on the battlefield.

**Comment by oleg-derevenetz** [comment]
I think that if player left the battle log in expanded state, then he knows what he's doing, and it is wrong to think a priori that he is doing something wrong. For example, he may want to follow the exact AI actions during the AI turn in the realtime.

**Reply by Districh-ru** [reply]
So then we may have an issue that the battle log cannot be opened/closed while performing any animation or a sequence of actions, like auto-battle mode or sequence of enemy moves.

**Reply by oleg-derevenetz** [reply]
Why is this an issue? While the user is not in control, he cannot do many things except toggling the battle log. He cannot view hero's properties or creature properties, he cannot open battle options window. Why toggling the battle log is different among these things?

**Reply by Districh-ru** [reply]
The action on the battlefield cannot be performed with opened hero's or creature's properties, options dialog, spell book. This elements are always off in this case and do not cover the the battlefield. The battle log could be left opened and the user may want to watch the action, that began behind it.

**Reply by oleg-derevenetz** [reply]
He may as well want to watch the battle log instead of visual effects behind it. We never know this in advance, so I believe that we should give the user some choice and let him decide for himself. He may also be annoyed by the inability to change the speed of the currently walking unit, there is no big difference IMHO :)

**Reply by zenseii** [reply]
So basically it should be possible to show/hide the battle-log during the AI's turn and during auto-combat, is what you are saying. This is probably better than auto-hiding it for the reasons explained here.

**Reply by oleg-derevenetz** [reply]
> So basically it should be possible to show/hide the battle-log during the AI's turn and during auto-combat, is what you are saying.

This is basically an equivalent of "should be possible to show/hide the battle log during the ongoing unit movements and spell animations", because that's what takes up most of the time during the AI turn or auto combat. The time directly spent by AI on making a decision is so short that it can be neglected.

**Reply by Districh-ru** [reply]
Yes, it would be good to be able to show/hide the combat log during unit/spell animation. And it's better than forced auto-hide that I suggested earlier.
The ability to open the battle options at any time and change some settings like animation speed sounds good, but at the moment it's not as important as fixing existing game bugs and making game optimizations, IMHO.
The user may want to do a lot of things during the AI action animation, and I believe that in the future we can make the game more flexible, like it is made in "HoMM3 HD" (skip long pre-battle sound, exit battle to load previous save, etc.). :)
I wrote about the battle log because there was a case when I opened it (for debugging), did some actions with visible units and then the enemy started a sequence of moves of his units and some of them were hidden by the combat log, which I could not close. This case taught me to always hide the battle log. :)
